src/jv.h

```diff
@@ -5,6 +5,20 @@
 #include <stdint.h>
 #include <stdio.h>
 
+#if defined(DISPATCH_FALLTHROUGH)
+# define JQ_FALLTHROUGH DISPATCH_FALLTHROUGH
+#else
+# if defined(__GNUC__)
+#  if __GNUC__ >= 7
+#   define JQ_FALLTHROUGH __attribute__((fallthrough))
+#  else
+#   define JQ_FALLTHROUGH do {} while (0) /* fallthrough */
+#  endif
+# else
+#  define JQ_FALLTHROUGH do {} while (0) /* fallthrough */
```

Comment:
I found `fallthrough` macro in Linux code and the code using it looks nice, but should we use JQ_ prefix to avoid conflict?

Reply:
@itchyny well, we need one that is portable to not-Linux, including BSDs, OS X, and Windows, as well as different compilers.  I did struggle with: what header file to put this in, and what to call this macro.

It's very annoying that there isn't a universally available way to do this...